support/cas-server-support-syncope-authentication/src/main/java/org/apereo/cas/syncope/authentication/SyncopeAuthenticationHandler.java

```diff
@@ -124,14 +117,24 @@ public SyncopeAuthenticationHandler(final String name,
 
         if (user.has("relationships")) {
             user.get("relationships").forEach(r -> attributes.put(
-                "syncopeUserRelationships" + r.get("type").asText(),
-                List.of(r.get("otherEndName").asText())));
+                    "syncopeUserRelationships" + r.get("type").asText(),
+                    List.of(r.get("otherEndName").asText())));
         }
 
         if (user.has("plainAttrs")) {
             user.get("plainAttrs").forEach(a -> attributes.put(
-                "syncopeUserAttr" + a.get("schema").asText(),
-                MAPPER.convertValue(a.get("values"), ArrayList.class)));
+                    "syncopeUserAttr_" + a.get("schema").asText(),
+                    MAPPER.convertValue(a.get("values"), ArrayList.class)));
+        }
+        if (user.has("derAttrs")) {
+            user.get("derAttrs").forEach(a -> attributes.put(
+                    "syncopeUserAttr_" + a.get("schema").asText(),
+                    MAPPER.convertValue(a.get("values"), ArrayList.class)));
+        }
+        if (user.has("virAttrs")) {
+            user.get("virAttrs").forEach(a -> attributes.put(
+                    "syncopeUserAttr_" + a.get("schema").asText(),
+                    MAPPER.convertValue(a.get("values"), ArrayList.class)));
```

Comment:
This is all excellent. For all new lines, you should additionally modify the related test for the handler to make sure these are covered.